components/output-mapper/src/main/java/org/wso2/siddhi/extension/output/mapper/wso2event/WSO2SinkMapper.java

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.siddhi.extension.output.mapper.wso2event;
+
+import org.apache.log4j.Logger;
+import org.wso2.carbon.databridge.commons.AttributeType;
+import org.wso2.carbon.databridge.commons.exception.MalformedStreamDefinitionException;
+import org.wso2.siddhi.annotation.Example;
+import org.wso2.siddhi.annotation.Extension;
+import org.wso2.siddhi.annotation.Parameter;
+import org.wso2.siddhi.annotation.util.DataType;
+import org.wso2.siddhi.core.event.Event;
+import org.wso2.siddhi.core.exception.ConnectionUnavailableException;
+import org.wso2.siddhi.core.stream.output.sink.SinkListener;
+import org.wso2.siddhi.core.stream.output.sink.SinkMapper;
+import org.wso2.siddhi.core.util.config.ConfigReader;
+import org.wso2.siddhi.core.util.transport.DynamicOptions;
+import org.wso2.siddhi.core.util.transport.OptionHolder;
+import org.wso2.siddhi.core.util.transport.TemplateBuilder;
+import org.wso2.siddhi.extension.output.mapper.wso2event.service.StreamDefinitionHolder;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.definition.Attribute.Type;
+import org.wso2.siddhi.query.api.definition.StreamDefinition;
+import org.wso2.siddhi.query.api.exception.ExecutionPlanValidationException;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+
+/**
+ * Mapper class to convert a Siddhi message to a WSO2 event. We will be using the stream definition and populate the
+ * WSO2 event attribute values accordingly and construct the WSO2 event. In case of null values, Same will be added
+ * to the WSO2 event attributes as well.
+ */
+@Extension(
+        name = "wso2event",
+        namespace = "sinkMapper",
+        description = "Event to WSO2 output mapper. Transports which publish WSO2 messages, can utilize this " +
+                "extension to convert the Siddhi event to WSO2 event objects. Users can send pre-defined WSO2 event " +
+                "format which adheres from the defined stream.",
+        parameters = {
+                @Parameter(name = "wso2event.stream.id",
+                        description =
```

Comment:
Added the forgotten parameter arbitrary.map.
The examples were already in the latter part of the extension description